webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -191,8 +200,13 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
       report.heading = this.getReportHeading(form, report);
       report.summary = form ? form.title : report.form;
       report.lineage = report.subject && report.subject.lineage || report.lineage;
+      // filter out the lineage level that belongs to the online logged in user
+      if(!this.sessionService.isOnlineOnly()) {
+        report.lineage = report.lineage.filter((level) => {
+          return (level !== this.currentLevel);
+        });
```

Comment:
That being said, will the user's currentLevel always be the last element in the lineage?  I think so, or the existing behavior of this would be kind of weird.

If that is the case, then it would be cheaper to just focus on the last element instead of looping through all of them...

```js
        if(report.lineage.length && report.lineage[length - 1] === this.currentLevel) {
          report.lineage.pop();
        }
```

Reply:
Yeah I felt the filter a bit expensive, thanks for the suggestion, then I guess it would make this line finally looking like 
```
      if (this.currentLevel && report.lineage.length && report.lineage[length - 1] === this.currentLevel) {
        report.lineage.pop();
      }
```

Reply:
Yeah you're right, I also felt using the filter a bit expensive but during my tests the size of the lineage array is not always what we would expected : some lineages look like `["CHW Betty's Area", null, null, null, null, null]`

Reply:
Oh weird!  I spend a bunch of time digging and trying to understand all the code that loaded the lineage data and still I do not feel like I have a solid grasp on how that data is produced.  I do not anticipate these lineage arrays being very large one way or the other, so there is really no performance concern here. I am fine with using `filter` here if that seems safer than `pop`.